### PR TITLE
[prometheus-postgres-exporter] Migrate to chart v2

### PR DESCRIPTION
**Pull request by zanhsieh**
See: https://github.com/prometheus-community/helm-charts/pull/390
Signed-off-by: zanhsieh <zanhsieh@gmail.com>

#### What this PR does / why we need it:
Initial attempt to move to chart v2, since helm v2 has been out of support from Nov. 15th, 2020.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-postgres-exporter]`)

### CODE REVIEW

File: charts/prometheus-postgres-exporter/templates/NOTES.txt [hunk 1]
```diff
@@ -1,3 +1,6 @@
+#################################################################################
```
Comment (monotek) [inline charts/prometheus-postgres-exporter/templates/NOTES.txt:1]: As the chart will not work in helm 2 i guess this notifications will not be shown to a helm 2 user anyway.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-postgres-exporter/templates/NOTES.txt (hunk 1 above)
Comment (zanhsieh) [reply charts/prometheus-postgres-exporter/templates/NOTES.txt:1]: Er... @monotek I think previously @scottrigby would like to have 'crystal clear' on supporting helm3. I don't know what else / where else I could put to make it so. What would you guys suggest? cc @scottrigby

Comment (monotek) [reply charts/prometheus-postgres-exporter/templates/NOTES.txt:1]: I think it's about to put it in the upgrading section of the chart.

Comment (scottrigby) [reply charts/prometheus-postgres-exporter/templates/NOTES.txt:1]: Yep, this is a suggestion not a demand. But it came from trial and error in the helm/charts repo. See this note in the [review guidelines](https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md#versioning) there:

> Any breaking (backwards incompatible) changes to a chart should:
> 1. Bump the MAJOR version
> 2. In the README, under a section called "Upgrading", describe the manual steps necessary to upgrade to the new (specified) MAJOR version

This came from many conversations about upgrading confusion, which lead to this issue https://github.com/helm/charts/issues/5657 and ultimately https://github.com/helm/charts/pull/11207.

Personally I'd suggest adopting the same policy for this repo too. Just to make it clear to end users what upgrading to each MAJOR version of a chart means, without them having to do some version diff of the requirements section in the chart README 😛

**Comment by zanhsieh** [comment]
@monotek @scottrigby 
I update the README.md for major version upgrade instruction. Please let me know anything else I need to do.

**Review by monotek — CHANGES_REQUESTED** [review]
please fix linter

File: charts/prometheus-postgres-exporter/README.md [hunk 2]
```diff
@@ -16,11 +21,8 @@ _See [helm repo](https://helm.sh/docs/helm/helm_repo/) for command documentation
 ## Install Chart
 
 ```console
-# Helm 3
+# Helm
```
Comment (scottrigby) [inline charts/prometheus-postgres-exporter/README.md:24]: I would just remove this comment
```suggestion
```

File: charts/prometheus-postgres-exporter/README.md [hunk 3]
```diff
@@ -30,21 +32,29 @@ _See [helm install](https://helm.sh/docs/helm/helm_install/) for command documen
 ## Uninstall Chart
 
 ```console
-# Helm 3
+# Helm
```
Comment (scottrigby) [inline charts/prometheus-postgres-exporter/README.md:35]:
```suggestion
```

File: charts/prometheus-postgres-exporter/README.md [hunk 4]
```diff
@@ -55,9 +65,6 @@ _See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documen
 See [Customizing the Chart Before Installing](https://helm.sh/docs/intro/using_helm/#customizing-the-chart-before-installing). To see all configurable options with detailed comments, visit the chart's [values.yaml](./values.yaml), or run these configuration commands:
 
 ```console
-# Helm 2
-$ helm inspect values prometheus-community/prometheus-postgres-exporter
-
-# Helm 3
+# Helm
```
Comment (scottrigby) [inline charts/prometheus-postgres-exporter/README.md:68]:
```suggestion
```

File: charts/prometheus-postgres-exporter/README.md [hunk 5]
```diff
@@ -30,21 +32,29 @@ _See [helm install](https://helm.sh/docs/helm/helm_install/) for command documen
...
 ```
 
 This removes all the Kubernetes components associated with the chart and deletes the release.
 
 _See [helm uninstall](https://helm.sh/docs/helm/helm_uninstall/) for command documentation._
 
-## Upgrading Chart
+## Upgrading
+
+### To 2.0.0
```
Comment (scottrigby) [inline charts/prometheus-postgres-exporter/README.md:45]: I would be very explicit here that the primary change in `2.0.0` is the Chart API `v1` to `v2`, which now requires Helm 3.

File: charts/prometheus-postgres-exporter/README.md [hunk 6]
```diff
@@ -30,21 +32,29 @@ _See [helm install](https://helm.sh/docs/helm/helm_install/) for command documen
...
 
 ```console
-# Helm 3 or 2
+$ kubectl patch deployment prometheus-postgres-exporter --type=json -p='[{"op": "remove", "path": "/spec/selector/matchLabels/chart"}]'
+```
+
+### Other minor version upgrade
+
+```console
+# Helm
```
Comment (scottrigby) [inline charts/prometheus-postgres-exporter/README.md:57]:
```suggestion
```

**Review by scottrigby — CHANGES_REQUESTED** [review]

**Comment by scottrigby** [comment]
> please fix linter

This is the markdownlint error https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md014---dollar-signs-used-before-commands-without-showing-output

**Review by monotek — APPROVED** [review]

**Comment by zanhsieh** [comment]
@scottrigby 
Please review again. Thank you.

**Review by scottrigby — APPROVED** [review]
Lgtm

**Comment by zanhsieh** [comment]
@gianrubio 
Please review. Thank you.

**Pull request closed without merging**